config/initializers/non_production_interceptor.rb

```diff
@@ -0,0 +1,6 @@
+app_domain = ENV.fetch("GOVUK_APP_DOMAIN", "")
+non_production_environment = %w(integration staging).detect { |s| app_domain.include?(s) }
+
+if non_production_environment
+  ActionMailer::Base.register_interceptor(NonProductionInterceptor)
```

Comment:
I've gone for `MailRecipientInterceptor` - I think this make sense because it's the recipient we're interested in.  We're not mentioning override anywhere so I think we can get away with not including that concept in the name.